stingray/io.py

```diff
@@ -92,108 +131,218 @@ def _get_additional_data(lctable, additional_columns):
+        instr = "unknown"
+    try:
+        mission = header["TELESCOP"].strip().lower()
+    except Exception:
+        mission = "unknown"
 
     ev_list += timezero
 
     # Read TSTART, TSTOP from header
     try:
-        t_start = np.longdouble(lchdulist[1].header['TSTART'])
-        t_stop = np.longdouble(lchdulist[1].header['TSTOP'])
-    except:  # pragma: no cover
-        logging.warning("Tstart and Tstop error. using defaults")
+        t_start = np.longdouble(header["TSTART"])
+        t_stop = np.longdouble(header["TSTOP"])
+    except Exception:  # pragma: no cover
+        warnings.warn(
+            "Tstart and Tstop error. using defaults", AstropyUserWarning
+        )
         t_start = ev_list[0]
         t_stop = ev_list[-1]
 
+    mjdref = np.longdouble(high_precision_keyword_read(header, "MJDREF"))
+
     # Read and handle GTI extension
-    accepted_gtistrings = gtistring.split(',')
+    accepted_gtistrings = gtistring.split(",")
 
     if gti_file is None:
         # Select first GTI with accepted name
         try:
-            gti_list = \
-                _get_gti_from_extension(
-                    lchdulist, accepted_gtistrings=accepted_gtistrings)
-        except:  # pragma: no cover
-            warnings.warn("No extensions found with a valid name. "
-                          "Please check the `accepted_gtistrings` values.")
-            gti_list = np.array([[t_start, t_stop]],
-                                dtype=np.longdouble)
+            gti_list = get_gti_from_all_extensions(
+                lchdulist,
+                accepted_gtistrings=accepted_gtistrings,
+                det_numbers=det_number,
+            )
+        except Exception:  # pragma: no cover
+            warnings.warn(
+                "No extensions found with a valid name. "
+                "Please check the `accepted_gtistrings` values.",
+                AstropyUserWarning,
+            )
+            gti_list = np.array([[t_start, t_stop]], dtype=np.longdouble)
     else:
         gti_list = load_gtis(gti_file, gtistring)
 
+    if additional_columns is None:
+        additional_columns = ["PI"]
+    if "PI" not in additional_columns:
+        additional_columns.append("PI")
+    if "PHA" not in additional_columns and mission.lower() in ["swift", "xmm"]:
+        additional_columns.append("PHA")
+
     additional_data = _get_additional_data(lctable, additional_columns)
 
     lchdulist.close()
 
     # Sort event list
     order = np.argsort(ev_list)
     ev_list = ev_list[order]
+    if detector_id is not None:
+        detector_id = detector_id[order]
 
     additional_data = order_list_of_arrays(additional_data, order)
 
+    pi = additional_data["PI"].astype(np.float32)
+    cal_pi = pi
+
+    if mission.lower() in ["xmm", "swift"]:
+        pi = additional_data["PHA"].astype(np.float32)
+        cal_pi = additional_data["PI"]
+        # additional_data.pop("PHA")
+
+    # additional_data.pop("PI")
+
     returns = EvtData()
```

Comment:
I am curious why there's a separate class `EvtData()` here if we also have a more built-out class `EventList`? Why not use the latter? I worry that having those two very similarly names classes might lead to confusion.

Reply:
This contains a lot more than EventList, it's basically an object containing all the information that we might or might not use to create an EventList afterwise.
I thought this function as a lower level file reader rather than a way to get EventLists straight from a FITS file

Reply:
So are you expecting this to be used by the user, or just by internal code? If it's the latter, it should probably get an underscore? If it's a user-exposed function, I think my argumentation hasn't really changed all that much: We have something called `EventList`, and here we introduce something called `EventData`, and if I were a user, and not careful about remembering which is named what and used in which context, I think I'd be confused and end up with inevitable errors. Especially since we *also* have documented functionality for loading an `EventList` object from a fits file. You could just turn this into a dictionary to make the issue go away?

Reply:
Good point. I called it EventReadOutput